### PR TITLE
Upgrade to Lit 2

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

NA

**Description**

A new major version of Lit has been announced a few weeks ago, see https://lit.dev/blog/2021-04-21-lit-2.0-meet-lit-all-over-again for details. It promises smaller bundles, better performances and up to date code. 

To keep our ui-component library up to date I upgraded to Lit 2 rc and fix the breaking changes according to the docs.

--

Update: Lit 2.0 stable has been released: https://lit.dev/blog/2021-09-21-announcing-lit-2 so I updated this PR with the new version and so we will be good to go 🚀

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,6 +110,7 @@
     "jest": "^26.4.2",
     "license-check-and-add": "^4.0.2",
     "lint-staged": "^10.5.4",
+    "lit-html": "^2.0.0-rc.3",
```
Comment (gaetanmaisse) [inline package.json:113]: Needed by `@storybook/web-components` for now

File: src/atoms/gv-icon.js [hunk 2]
```diff
@@ -13,9 +13,20 @@
...
-import { css, LitElement, html } from 'lit-element';
-import { directive } from 'lit-html';
-import { classMap } from 'lit-html/directives/class-map';
+import { css, LitElement, html } from 'lit';
+import { Directive, directive } from 'lit/directive';
+
+import { classMap } from 'lit/directives/class-map';
+import { noChange } from 'lit-html';
+
+class Namespaced extends Directive {
```
Comment (gaetanmaisse) [inline src/atoms/gv-icon.js:22]: New version of the Directive in Lit 2

File: stories/policy-studio/gv-policy-studio.test.js [hunk 3]
```diff
@@ -151,10 +151,10 @@ describe('P O L I C Y  S T U D I O', () => {
...
       expect(component._tabId).toEqual('design');
 
-      component.updateComplete.then(() => {
-        component._onAddFlowPlan({ detail: { planIndex: 1 } });
-
-        component.updateComplete.then(() => {
+      component.updateComplete
+        .then(() => component._onAddFlowPlan({ detail: { planIndex: 1 } }))
+        .then(() => component.updateComplete)
+        .then(() => {
```
Comment (gaetanmaisse) [inline stories/policy-studio/gv-policy-studio.test.js:157]: Fix flaky test

File: src/mixins/input-element.js [hunk 4]
```diff
@@ -111,7 +111,7 @@ export function InputElement(ParentClass) {
     }
 
     getInputElement() {
-      return this.shadowRoot.querySelector('input');
+      return this.shadowRoot?.querySelector('input');
```
Comment (gaetanmaisse) [inline src/mixins/input-element.js:114]: Lit is now too fast to render and sometime the shadowRoot isn't ready yet

**Review by jgiovaresco — APPROVED** [review]
💪

File: package.json [hunk 5]
```diff
@@ -62,8 +62,7 @@
     "highlight.js": "^10.4.0",
     "jdenticon": "^2.2.0",
     "jsonschema": "^1.4.0",
-    "lit-element": "^2.5.1",
-    "lit-html": "^1.4.1",
+    "lit": "^2.0.0-rc.2",
```
Comment (loriepisicchio) [inline package.json:65]: :scream:  seeing this RC dependency makes me kind of nervous :smile:

**Review by loriepisicchio — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (gcusnieux) [reply package.json:64]: same for me

**Review by gcusnieux — APPROVED** [review]

**Pull request merged**